tests/unit/openssl_lib_mock.h

```diff
@@ -35,6 +35,13 @@ namespace openssl
 class OpenSSLLibMockInterface
 {
 public:
+    virtual const char* SSL_EC_curve_nid2nist(int nid) = 0;
```

Comment:
I agree here. We currently don't have any tests for the HSM code except for the example - which (iirc) also executed as a test but maybe a bit misleading.
This is actually a good place where this mock layer here comes in handy. You can actually write a test for your HSM class by mocking the openssl interface and then checking that is submits the right call to SSL_ENGINE_ctrl_cmd and even verify if the parameters look sane-ish.

Reply:
Done-ish